test/HeadersTest.php

```diff
@@ -424,4 +424,22 @@ public function testHeaderCrLfAttackAddHeadersMultiple()
         $headers->addHeaders(['Fake' => ["foo-bar\r\n\r\nevilContent"]]);
         $headers->forceLoading();
     }
+    
+    public function testCanConvertNonASCIIHeaderToStringAfterAddingHeaderToMessage()
+    {
+        //non ASCII characters
+        $str = 'òàù';
+        $header = new \Zend\Mail\Header\GenericHeader();
+        $message = new \Zend\Mail\Message();
+        $headers = new \Zend\Mail\Headers();
+        
+        $header->setFieldName('X-Test');
+        $header->setFieldValue($str);
+        $headers->addHeader($header);
+        $message->setHeaders($headers);
+        $headers->setEncoding('UTF-8');
+        
+		$this->assertTrue(is_string($header->toString()), 'Can call Header::toString()');
```

Comment:
The problem is not an incorrect encoding, but just the ability to add non ascii headers to a mail message.  The goal of the test is to show that If you create a Zend\Mail\Header() with non ascii value, you can't call toString Method after you add it to a Zend\Mail\Message().
I really don't have good experience in writing tests, I know!